php/libraries/CouchDB.class.inc

```diff
@@ -610,9 +610,9 @@ class CouchDB
      *
      * @param array $doc A PHP array encoding the JSON document to be created
      *
-     * @return string the JSON string returned by the CouchDB server
+     * @return ?string the JSON string returned by the CouchDB server
      */
-    function postDoc(array $doc): string
+    function postDoc(array $doc): ?string
```

Comment:
This shouldn't be nullable. The CouchDB response should always be valid JSON.

Reply:
This issue comes from `_postRelativeURL` which has a nullable return type because `json_decode` itself can return null in some cases. 

Should we throw an Exception here (and below) if `$json` is null?

Reply:
the response can be: 1. empty (in which the empty string should be returned) or 2. valid json. If json_decode fails, it should be an exceptional circumstance. We can probably get that behaviour by passing the JSON_THROW_ON_ERROR option to json_decode.